org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/UriExtensions.xtend

```diff
@@ -31,8 +33,8 @@ class UriExtensions {
 
 	def String toPath(java.net.URI uri) {
 		try {
-			val path = Paths.get(uri)
-			return path.toUri.toString
+			val path = Paths.get(uri.path)
+			return URLDecoder.decode(path.toUri.toString, "UTF-8");
```

Comment:
It is just a minor thing but what about using the `StandardCharsets.UTF_8.name` instead of the `"UTF-8"` string?